src/Service/LoggerService.php

```diff
@@ -0,0 +1,181 @@
+            'level' => self::ADYEN_CRONJOB,
+            'fileName' => 'adyen_cronjob.log'
+        ),
+        self::NOTICE => array(
+            'level' => self::NOTICE,
+            'fileName' => 'notice.log'
+        ),
+        self::WARNING => array(
+            'level' => self::WARNING,
+            'fileName' => 'warning.log'
+        ),
+        self::ERROR => array(
+            'level' => self::ERROR,
+            'fileName' => 'error.log'
+        ),
+    );
+
+    /**
+     * Logging levels from syslog protocol defined in RFC 5424
+     * Overrule the default to add Adyen specific loggers to log into separate files
+     *
+     * @var array $levels Logging levels
+     */
+    protected static $levels = array(
+        self::DEBUG => 'DEBUG',
+        self::INFO => 'INFO',
+        self::ADYEN_API => 'ADYEN_API',
+        self::ADYEN_RESULT => 'ADYEN_RESULT',
+        self::ADYEN_NOTIFICATION => 'ADYEN_NOTIFICATION',
+        self::ADYEN_CRONJOB => 'ADYEN_CRONJOB',
+        self::NOTICE => 'NOTICE',
+        self::WARNING => 'WARNING',
+        self::ADYEN_ERROR => 'ADYEN_ERROR',
+        self::CRITICAL => 'CRITICAL',
+        self::ALERT => 'ALERT',
+        self::EMERGENCY => 'EMERGENCY',
+    );
+
+    public function __construct() {
+        parent::__construct(self::NAME);
+        $this->registerAdyenLogHandlers();
+        $this->pushProcessor((new PsrLogMessageProcessor()));
+    }
+
+    /**
+     * Retrieve Adyen log path
+     * If it doesn't exist yet then also creates it
+     *
+     * @return string
+     * @throws CommandException
+     */
+    private function getAdyenLogPath()
+    {
+        $path = self::SHOPWARE_LOG_PATH . self::LOG_DIR;
+
+        if (!file_exists($path)) {
+            if (!mkdir($path, 0755, true)) {
+                throw new CommandException('Creating the Adyen log folder failed');
+            }
+        }
+
+        return $path;
+    }
+
+    /**
+     * @param string $message
+     * @param array $context
+     * @return bool
+     */
+    public function addAdyenAPI($message, array $context = array())
+    {
+        return $this->addRecord(static::ADYEN_API, $message, $context);
+    }
+
+    /**
+     * @param string $message
+     * @param array $context
+     * @return bool
+     */
+    public function addAdyenResult($message, array $context = array())
+    {
+        return $this->addRecord(static::ADYEN_RESULT, $message, $context);
+    }
+
+    /**
+     * @param string $message
+     * @param array $context
+     * @return bool
+     */
+    public function addAdyenNotification($message, array $context = array())
+    {
+        return $this->addRecord(static::ADYEN_NOTIFICATION, $message, $context);
+    }
+
+    /**
+     * @param string $message
+     * @param array $context
+     * @return bool
+     */
+    public function addAdyenCronjob($message, array $context = array())
+    {
+        return $this->addRecord(static::ADYEN_CRONJOB, $message, $context);
+    }
```

Comment:
Shouldn't these methods be called `logSomething`? or just `something` for consistency with PSR-3?

Reply:
Agree. Let's go for the `something` :)